adcm_aio_client/core/filters.py

```diff
@@ -10,5 +10,143 @@
+
+            query[f"{name}__{operation}"] = value
+
+        return query
+
+    def _check_allowed(self: Self, filter_: Filter) -> None:
+        allowed_filter = self._allowed.get(filter_.attr)
+        if not allowed_filter:
+            message = f"Filter by {filter_.attr} is not allowed. Allowed: {', '.join(self._allowed)}"
+            raise InvalidFilterError(message)
+
+        if filter_.op not in allowed_filter.operations:
+            message = f"Operation {filter_.op} is not allowed. Allowed: {', '.join(sorted(allowed_filter.operations))}"
+            raise InvalidFilterError(message)
+
+        expected_type = allowed_filter.single_input
+        if isinstance(filter_.value, Iterable):
+            if not all(isinstance(entry, expected_type) for entry in filter_.value):
+                message = f"At least one entry is not {expected_type}: {filter_.value}"
+                raise InvalidFilterError(message)
+        else:
+            if not isinstance(filter_.value, expected_type):
+                message = f"Value {filter_.value} is not {expected_type}"
+                raise InvalidFilterError(message)
+
+    def _attribute_name_to_camel_case(self: Self, name: str) -> str:
+        first, *rest = name.split("_")
+        return f"{first}{''.join(map(str.capitalize, rest))}"
+
+    def _simplify_value(self: Self, value: FilterValue) -> SimplifiedValue:
+        if isinstance(value, (str, int)):
+            return value
+
+        if isinstance(value, InteractiveObject):
+            return value.id
+
+        simplified_collection = deque()
+
+        for entry in value:
+            if isinstance(entry, (str, int)):
+                simplified_collection.append(entry)
+            elif isinstance(entry, InteractiveObject):
+                simplified_collection.append(entry.id)
+            else:
+                message = f"Failed to simplify: {entry}"
+                raise TypeError(message)
+
+        return tuple(simplified_collection)
+
+    def _check_no_operation_value_conflict(self: Self, operation: str, value: SimplifiedValue) -> None:
+        is_collection = isinstance(value, tuple)
+
+        if operation in MULTI_OPERATIONS:
+            if not is_collection:
+                message = f"Multiple values expected for {operation}"
+                raise InvalidFilterError(message)
+
+            if not value:
+                message = "Collection for filter shouldn't be empty"
+                raise InvalidFilterError(message)
+
+        else:
+            if is_collection:
+                message = f"Only one value is expected for {operation}"
+                raise InvalidFilterError(message)
+
+    def _prepare_query_param_value(self: Self, value: SimplifiedValue) -> str:
+        if isinstance(value, tuple):
+            return ",".join(map(str, value))
+
+        return str(value)
+
+
+FilterByName = FilterBy("name", ALL_OPERATIONS, str)
+FilterByDisplayName = FilterBy("display_name", ALL_OPERATIONS, str)
```

Comment:
What do you think if we don't give users the opportunity to use names with underscores, and force them to use names in the camelcase style?
I understand the disadvantages, you will have to use named arguments in methods not according to PEP8, for example `.filter(displayName__eq="service")`

Reply:
I am against that, because field in objects are `display_name`, etc.

it's counter-intuitive

Reply:
I agree